tests/extras/datasets/api/test_api.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2020 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=no-member,unused-argument
+import socket
+import unittest
+
+import pytest
+import requests
+import requests_mock
+
+from kedro.extras.datasets.api import APIDataSet
+from kedro.io.core import DataSetError, DataSetNotFoundError
+
+POSSIBLE_METHODS = ["GET", "OPTIONS", "HEAD", "POST", "PUT", "PATCH", "DELETE"]
+
+TEST_URL = "http://example.com/api/test"
+TEST_TEXT_RESPONSE_DATA = "This is a response."
+TEST_JSON_RESPONSE_DATA = [{"key": "value"}]
+
+TEST_PARAMS = {"param": "value"}
+TEST_URL_WITH_PARAMS = TEST_URL + "?param=value"
+
+TEST_HEADERS = {"key": "value"}
+
+
+@requests_mock.Mocker()
+class TestAPIDataSet(unittest.TestCase):
+    def test_successfully_load_with_text_response(self, requests_mocker):
+        for method in POSSIBLE_METHODS:
```

Comment:
Can you please use `pytest.mark.parametrize` as in [test_biosequence_dataset](https://github.com/quantumblacklabs/kedro/blob/9809bd7ca0556531fa4a2fc02d5b2dc26cf8fa97/tests/extras/datasets/bioinformatics/test_biosequence_dataset.py#L94), for example?

Reply:
This is a neat trick !